core/src/main/java/org/elasticsearch/index/snapshots/blobstore/BlobStoreIndexShardRepository.java

```diff
@@ -359,38 +359,33 @@ public BlobStoreIndexShardSnapshot loadSnapshot() {
          */
         protected void finalize(List<SnapshotFiles> snapshots, int fileListGeneration, Map<String, BlobMetaData> blobs) {
             BlobStoreIndexShardSnapshots newSnapshots = new BlobStoreIndexShardSnapshots(snapshots);
-            List<String> blobsToDelete = new ArrayList<>();
             // delete old index files first
-            for (String blobName : blobs.keySet()) {
-                // delete old file lists
+            blobs.keySet().forEach(blobName -> {
                 if (indexShardSnapshotsFormat.isTempBlobName(blobName) || blobName.startsWith(SNAPSHOT_INDEX_PREFIX)) {
-                    blobsToDelete.add(blobName);
+                    try {
+                        blobContainer.deleteBlob(blobName);
+                    } catch (IOException e) {
+                        // We cannot delete index file - this is fatal, we cannot continue, otherwise we might end up
+                        // with references to non-existing files
+                        throw new IndexShardSnapshotFailedException(shardId, "error deleting index file ["
+                            + blobName + "] during cleanup", e);
+                    }
                 }
-            }
-
-            try {
-                blobContainer.deleteBlobs(blobsToDelete);
-            } catch (IOException e) {
-                // We cannot delete index file - this is fatal, we cannot continue, otherwise we might end up
-                // with references to non-existing files
-                throw new IndexShardSnapshotFailedException(shardId, "error deleting index files during cleanup", e);
-            }
+            });
 
-            blobsToDelete = new ArrayList<>();
-            // now go over all the blobs, and if they don't exists in a snapshot, delete them
-            for (String blobName : blobs.keySet()) {
+            // now go over all the blobs, and if they don't exist in a snapshot, delete them
+            blobs.keySet().forEach(blobName -> {
                 // delete unused files
                 if (blobName.startsWith(DATA_BLOB_PREFIX)) {
                     if (newSnapshots.findNameFile(FileInfo.canonicalName(blobName)) == null) {
-                        blobsToDelete.add(blobName);
+                        try {
+                            blobContainer.deleteBlob(blobName);
+                        } catch (IOException e) {
+                            logger.debug("[{}] [{}] error deleting blob [{}] during cleanup", e, snapshotId, shardId, blobName);
```

Comment:
Why are we swallowing this? I know that's the way that it was before, but is it right?